### PR TITLE
upgrade clap to 4.0

### PR DESCRIPTION
**Pull request by uran0sH**
In order to use derive API to refactor, we upgrade clap firstly.

Signed-off-by: Wenyu Huang <huangwenyuu@outlook.com>

### CODE REVIEW

**Comment by imeoer** [comment]
@uran0sH Sorry, please rebase master first.

**Review by jiangliu — APPROVED** [review]

File: src/bin/nydus-image/main.rs [hunk 1]
```diff
@@ -366,202 +345,197 @@ fn prepare_cmd_args(bti_string: String) -> ArgMatches<'static> {
...
                         .help("path to RAFS metadata blob (required)")
-                        .required(true)
-                        .takes_value(true),
+                        .required(true),
                 )
                 .arg(
-                    Arg::with_name("verbose")
+                    Arg::new("verbose")
                         .long("verbose")
-                        .short("v")
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:389]: Should be `v` here?

File: src/bin/nydusctl/main.rs [hunk 2]
```diff
@@ -27,149 +27,147 @@ use commands::{
...
+lazy_static! {
+    static ref BTI: BuildTimeInfo = BuildTimeInfo::dump().1;
+}
+
 #[tokio::main]
 async fn main() -> Result<()> {
-    let (_, build_info) = BuildTimeInfo::dump();
-    let app = App::new("A client to query and configure the nydusd daemon\n")
-        .version(build_info.package_ver.as_str())
+    // let (_, build_info) = BuildTimeInfo::dump();
```
Comment (imeoer) [inline src/bin/nydusctl/main.rs:36]: Maybe remove it directly.

File: src/bin/nydus-image/main.rs [hunk 3]
```diff
@@ -366,202 +345,197 @@ fn prepare_cmd_args(bti_string: String) -> ArgMatches<'static> {
...
-fn main() -> Result<()> {
-    let (bti_string, build_info) = BuildTimeInfo::dump();
+lazy_static! {
+    static ref BTI_STRING: String = BuildTimeInfo::dump().0;
+    static ref BTI: BuildTimeInfo = BuildTimeInfo::dump().1;
+}
 
-    let cmd = prepare_cmd_args(bti_string);
+fn main() -> Result<()> {
+    // let (bti_string, build_info) = BuildTimeInfo::dump();
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:534]: Ditto.

**Review by imeoer — APPROVED** [review]

**Comment by jiangliu** [comment]
Failed to handle --prefetch-files option:
root@0fba9939116d:/nydus# target/debug/nydusd fuse -M mnt --log-level trace -C nydusd-config1.json -B images/c5016d92995cb3553bd62fc04229f52317d94ac5e76ed65b2a09928ba75fbbf3 --prefetch-files / &
[8] 63069
root@0fba9939116d:/nydus# [2022-10-30 17:06:48.702673 +00:00] INFO [app/src/lib.rs:74] Program Version: v2.1.0-rc.3.1-225-ge799bf75d, Git Commit: "e799bf75d28598291e9f6db4cbcb38be59379cf1", Build Time: "2022-10-30T17:00:17.704862586Z", Profile: "debug", Rustc Version: "rustc 1.61.0 (fe5b13d68 2022-05-18)"
[2022-10-30 17:06:48.702935 +00:00] INFO [src/bin/nydusd/main.rs:516] Set rlimit-nofile to 1000000, maximum 1048576
[2022-10-30 17:06:49.206767 +00:00] ERROR [/root/.cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.1.0/src/lib.rs:130] thread 'main' panicked at 'Mismatch between definition and access of `prefetch-files`. Could not downcast to &str, need to downcast to alloc::string::String
': src/bin/nydusd/main.rs:545
   0: <backtrace::capture::Backtrace as core::default::Default>::default
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.66/src/capture.rs:410:9
   1: log_panics::Config::install_panic_hook::{{closure}}
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.1.0/src/lib.rs:115:29
   2: std::panicking::rust_panic_with_hook
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:702:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:588:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/sys_common/backtrace.rs:138:18
   5: rust_begin_unwind
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/std/src/panicking.rs:584:5
   6: core::panicking::panic_fmt
             at /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/core/src/panicking.rs:143:14
   7: clap::parser::error::MatchesError::unwrap
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-4.0.18/src/parser/error.rs:30:9
   8: clap::parser::matches::arg_matches::ArgMatches::get_many

**Pull request merged**